### DISCUSSION TITLE
Automated translations with different keys

### DISCUSSION THREAD

**Discussion by dusan-smolnikar** [WeblateOrg/weblate#9099] [Q&A] [answered]
Hi,

I have a legacy project with translations that I need to keep maintaining. Translations are monolingual, in JSON files. Example:
```
"hompagelinkinnavigation" : "Home"
```

I am however working on a new version, which will live in parallel with the old one for some time. I would like to start the language files from a clean slate, using a more structured language file, i.e.:
```
"navigation": {
  "home": "Home"
}
```

I would like the new version to automatically fetch translations for all languages from the legacy project. I would imagine I need a mapper from old to new labels for example,
```
hompagelinkinnavigation = navigation.home
```

Is this possible through the automatic translation add-on?

Do you propose a different approach instead? While the old files need to remain unchanged, I can change the format of the new files if required.

**Comment by nijel** [comment]
You can use translation memory for, it does look at strings only.

**Reply by dusan-smolnikar** [reply]
So it will match by the translated string in the source language? Meaning I just need to make sure both english translations are the same, and the other languages will follow suit?

**Reply by nijel** [reply, marked as answer]
It will match source string against translation memory.

**Reply by dusan-smolnikar** [reply]
Thanks!
